Zend/zend_vm_def.h

```diff
@@ -1560,7 +1561,8 @@ ZEND_VM_HELPER(zend_pre_dec_helper, VAR|CV, ANY)
 		}
 		decrement_function(var_ptr);
 		if (UNEXPECTED(EG(exception))) {
-			/* Smart branch expects result to be set with exceptions */
+			FREE_OP1();
+			/* opcodes are expected to set the result value */
 			if (UNEXPECTED(RETURN_VALUE_USED(opline))) {
 				ZVAL_NULL(EX_VAR(opline->result.var));
 			}
```

Comment:
Are you sure you need all this code?
Removing the whole ``if (UNEXPECTED(EG(exception))) {...}`` shouldn't change anything.

Reply:
Don't think so no, I'm still figuring out the VM at times.

I think my rationale was that I wanted to deal with stuff localy to be sure I understand what is going on. But as nothing else happens here this code can be dropped.